### PR TITLE
[#180313529] Request verbose carrier errors

### PR DESCRIPTION
**Pull request by adamdullenty**
This card is to add the new [verbose carrier errors](https://docs.sendcloud.sc/api/v2/shipping/#create-parcels-with-receiving-carrier-error) flag to our Sendcloud Create Parcel API requests so we can record these errors as DeliveryProblems and display them to CS users

### Why

**As ** CD
**I want ** announcement errors from Sendcloud to be recorded as a DeliveryProblem
**So that ** I have the information on hand to fix the issue before re-locking

### Acceptance Criteria/s

```gherkin
Scenario: Creating a parcel
Given a delivery that is ready to lock
When we try to create a parcel for that delivery via Sendcloud's API
Then we include the `errors=verbose-carrier` flag to ensure that we get detailed validation errors from the carrier
```

```gherkin
Scenario: Creating a parcel with an announcement error
Given a delivery that is ready to lock
When we try to create a parcel for that delivery via Sendcloud's API
And the request fails with a 400 status response and a verbose carrier error
Then we store the validation error as a DeliveryProblem
```

### Associated Trello Card
-
### Who (or what) is affected / how could this be rolled out/ how will we track this?
-
### Ideas for testing (these usually come from developers)
1)
2)

### Notes

* Docs for the new flag: https://docs.sendcloud.sc/api/v2/shipping/#create-parcels-with-receiving-carrier-error

### Pivotal link

https://www.pivotaltracker.com/story/show/180313529

### CODE REVIEW

File: lib/sendcloud/errors/response_error.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def initialize(payload:, body:, status:)
       private
 
       def build_message
-        "#{status} #{body[:error][:message]}"
+        "Sendcloud error: #{status} - #{body[:error][:message]}"
```
Comment (adamdullenty) [inline lib/sendcloud/errors/response_error.rb:17]: Some of the errors can be a bit cryptic or refer to things like "please contact support", and we store these errors directly on the DeliveryProblem, so I added "Sendcloud" to the message to clarify where the error is actually coming from.

File: lib/sendcloud/operations/create_parcel_request.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def handle_response_body(body)
       end
 
       def endpoint
-        "parcels"
+        "parcels?errors=verbose-carrier"
```
Comment (adamdullenty) [inline lib/sendcloud/operations/create_parcel_request.rb:17]: This is the actual change

File: spec/sendcloud/operations/create_parcel_request_spec.rb [hunk 3]
```diff
@@ -68,7 +68,44 @@
...
+        }
+      end
+
+      it "raises an exception" do
+        VCR.use_cassette("create_parcel_request/invalid_request_with_carrier_errors") do
+          aggregate_failures do
+            expect {
+              subject.execute
+            }.to raise_error(Sendcloud::Errors::ResponseError,
+              'Sendcloud error: 400 - non_field_errors: "PostNL service error: \'1225029\'"')
```
Comment (gdackys) [inline spec/sendcloud/operations/create_parcel_request_spec.rb:108]: @adamdullenty, in this example, is `PostNL service error: '1225029'` what gets returned with the verbose flag switched on?

**Review by gdackys — APPROVED** [review]

File: spec/sendcloud/operations/create_parcel_request_spec.rb (hunk 3 above)
Comment (adamdullenty) [reply spec/sendcloud/operations/create_parcel_request_spec.rb:108]: Yes - this is the "verbose" response! I've asked Sendcloud for a description of all the service error codes we could receive.

Comment (adamdullenty) [reply spec/sendcloud/operations/create_parcel_request_spec.rb:108]: To be specific, `non_field_errors: "PostNL service error: \'1225029\'"` is what gets returned

**Pull request merged**